app/src/main/java/com/onboarding/wheatherapp/data/mapper/WeatherMapper.kt

```diff
@@ -2,22 +2,41 @@ package com.onboarding.wheatherapp.data.mapper
 
 import com.onboarding.wheatherapp.data.entity.Data
 import com.onboarding.wheatherapp.data.entity.Main
-import com.onboarding.wheatherapp.data.entity.Weather
+import com.onboarding.wheatherapp.data.entity.WeatherDescription
 import com.onboarding.wheatherapp.data.services.response.DataResponse
 import com.onboarding.wheatherapp.data.services.response.MainResponse
-import com.onboarding.wheatherapp.data.services.response.WeatherResponse
+import com.onboarding.wheatherapp.data.services.response.WeatherDescriptionResponse
+import java.text.SimpleDateFormat
+import java.util.ArrayList
+import java.util.Date
+import java.util.Locale
 
 class WeatherMapper {
 
     private fun transformMain(main: MainResponse) = Main(main.temp, main.tempMax, main.tempMin)
 
-    private fun transformList(list: ArrayList<DataResponse>): ArrayList<Data> {
+    private val apiFormat = SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.ENGLISH)
+    private val appFormat = SimpleDateFormat("EEEE, MM/dd", Locale.ENGLISH)
+
+    private fun transformWeatherDescription(weather: ArrayList<WeatherDescriptionResponse>): ArrayList<WeatherDescription> {
```

Comment:
why you are using a ArrayList ? please use a List or a MutableList in kotlin

Reply:
Solved!